pkg/network/ebpf/c/protocols/kafka/helpers.h

```diff
@@ -0,0 +1,298 @@
+    // Skipping number of entries for now
+    offset += sizeof(s32);
+
+    READ_BIG_ENDIAN_WRAPPER(s16, topic_name_size, skb, offset);
+    if (topic_name_size <= 0 || topic_name_size > TOPIC_NAME_MAX_STRING_SIZE) {
+        return false;
+    }
+
+    const u32 key = 0;
+    char *topic_name = bpf_map_lookup_elem(&kafka_topic_name, &key);
+    if (topic_name == NULL) {
+        return false;
+    }
+    bpf_memset(topic_name, 0, TOPIC_NAME_MAX_STRING_SIZE);
+
+    read_into_buffer_topic_name((char *)topic_name, skb, offset);
+    offset += topic_name_size;
+
+    CHECK_STRING_COMPOSED_OF_ASCII(TOPIC_NAME_MAX_STRING_SIZE, topic_name_size, topic_name);
+}
+
+// Trying to parse a produce request.
+static __always_inline bool try_parse_produce_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    const s16 api_version = kafka_header->api_version;
+
+    if (api_version >= 3) {
+        READ_BIG_ENDIAN_WRAPPER(s16, transactional_id_size, skb, offset);
+        if (transactional_id_size > 0) {
+            offset += transactional_id_size;
+        }
+    }
+
+    READ_BIG_ENDIAN_WRAPPER(s16, acks, skb, offset);
+    if (acks > 1 || acks < -1) {
+        // The number of acknowledgments the producer requires the leader to have received before considering a request
+        // complete. Allowed values: 0 for no acknowledgments, 1 for only the leader and -1 for the full ISR.
+        return false;
+    }
+
+    READ_BIG_ENDIAN_WRAPPER(s32, timeout_ms, skb, offset);
+    if (timeout_ms < 0) {
+        // timeout_ms cannot be negative.
+        return false;
+    }
+
+    return validate_first_topic_name(skb, offset);
+}
+
+// Trying to parse a fetch request.
+static __always_inline bool try_parse_fetch_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    // replica_id => INT32
+    // max_wait_ms => INT32
+    // min_bytes => INT32
+    offset += 3 * sizeof(s32);
+
+    if (kafka_header->api_version >= 7) {
+        // session_id => INT32
+        // session_epoch => INT32
+        offset += 2 * sizeof(s32);
+    }
+
+    if (kafka_header->api_version >= 4) {
+        // isolation_level => INT8
+        offset += sizeof(s8);
+    }
+
+    if (kafka_header->api_version >= 3) {
+        // max_bytes => INT32
+        offset += sizeof(s32);
+    }
+
+    return validate_first_topic_name(skb, offset);
+}
+
+// Calls the relevant function, according to the api_key.
+static __always_inline bool is_kafka_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    switch (kafka_header->api_key) {
+    case KAFKA_PRODUCE:
+        return try_parse_produce_request(kafka_header, skb, offset);
+    case KAFKA_FETCH:
+        return try_parse_fetch_request(kafka_header, skb, offset);
+    default:
+        return false;
+    }
+}
+
+// Checks if the packet represents kafka communication.
```

Comment:
This is relevant only for Kafka request. We should also add Kafka response if we want to classify all connections

Reply:
https://github.com/DataDog/datadog-agent/pull/15554#issuecomment-1438135600

Reply:
The comment is still misleading, consider changing it to "kafka request" and not "kafka communication" and say something about the response for now